### PR TITLE
ref(integrations): Move SLO metric tokens to tags

### PR DESCRIPTION
**Pull request by RyanSkonnord**
Change the structure for integration SLO metrics to move several tokens from the key to the tags. This will make certain kinds of monitoring easier.

Refactor EventLifecycleMetric to be more generic and extract IntegrationEventLifecycleMetric from it. The latter class is now more opinionated about key structure, replacing `get_standard_key` with abstract methods that enforce a consistent naming convention. Remove the unused "extra tokens" option.

This change will cause a discontinuity in metrics data when it is deployed.

### CODE REVIEW

File: src/sentry/integrations/utils/metrics.py [hunk 1]
```diff
@@ -25,65 +24,86 @@ def __str__(self) -> str:
...
+        raise NotImplementedError
 
-    def capture(self, assume_success: bool = True) -> "EventLifecycle":
-        """Open a context to measure the event."""
-        return EventLifecycle(self, assume_success)
+    def get_metric_id(self, outcome: EventLifecycleOutcome) -> MetricId:
+        """Construct the metrics key and tags that will represent this event."""
+
+        key_tokens = (
+            "sentry",
```
Comment (RyanSkonnord) [inline src/sentry/integrations/utils/metrics.py:95]: This shows up as "sentry.sentry.integrations.*" in Datadog so we probably want to remove this first "sentry" while we're making changes.

File: src/sentry/integrations/utils/metrics.py [hunk 2]
```diff
@@ -25,65 +24,86 @@ def __str__(self) -> str:
...
     """Information about an event to be measured.
 
-    This class is intended to be used across different integrations that share the
-    same business concern. Generally a subclass would represent one business concern
-    (such as MessagingInteractionEvent, which extends this class and is used in the
-    `slack`, `msteams`, and `discord` integration packages).
+    This is a generic base class not tied specifically to integrations. See
+    IntegrationEventLifecycleMetric for integration-specific key structure. (This
+    class could be moved from this module to a more generic package if we ever want
+    to use it outside of integrations.)
```
Comment (RyanSkonnord) [inline src/sentry/integrations/utils/metrics.py:39]: FYI @corps, since IIRC you expressed some interest in doing this.

Comment (corps) [reply src/sentry/integrations/utils/metrics.py:39]: Thank you!

File: src/sentry/integrations/utils/metrics.py (hunk 1 above)
Comment (RyanSkonnord) [reply src/sentry/integrations/utils/metrics.py]: Removed.

**Review by GabeVillalobos — APPROVED** [review]

**Pull request merged**